### PR TITLE
RemoteCache: Cleanup infra remote cache

### PR DESCRIPTION
**Pull request by Jguer**
**What is this feature?**

- Cleanup infra remote cache
- Add usage stats service
- Remove unused logger
- Add DO NOT USE for Count()
- Fix typo in error

### CODE REVIEW

**Comment by Jguer** [comment]
Added fix for extra issue highlighted by @kalleep on request context being used to cancel audit logging

File: pkg/infra/remotecache/remotecache.go [hunk 1]
```diff
@@ -44,12 +43,27 @@ func ProvideService(cfg *setting.Cfg, sqlStore db.DB, secretsService secrets.Ser
...
 
+func (ds *RemoteCache) getUsageStats(ctx context.Context) (map[string]interface{}, error) {
+	stats := map[string]interface{}{}
+	stats["stats.remote_cache."+ds.Cfg.RemoteCacheOptions.Name+".count"] = 1
+	encryptVal := 0
+	if ds.Cfg.RemoteCacheOptions.Encryption {
+		encryptVal = 1
+	}
+
+	stats["stats.remote_cache_encrypt.enabled.count"] = encryptVal
```
Comment (DanCech) [inline pkg/infra/remotecache/remotecache.go:62]:
```suggestion
	stats["stats.remote_cache.encrypt_enabled.count"] = encryptVal
```

**Review by DanCech — APPROVED** [review]
Looks good, left one small suggestion to make the stat naming more consistent

**Review by bergquist — APPROVED** [review]
LGTM

File: pkg/infra/remotecache/remotecache_test.go [hunk 2]
```diff
@@ -123,6 +124,24 @@ func canNotFetchExpiredItems(t *testing.T, client CacheStorage) {
 	assert.Equal(t, err, ErrCacheItemNotFound)
 }
 
+func TestCollectUsageStats(t *testing.T) {
+	wantMap := map[string]interface{}{
+		"stats.remote_cache.redis.count":           1,
+		"stats.remote_cache_encrypt.enabled.count": 1,
```
Comment (Jguer) [inline pkg/infra/remotecache/remotecache_test.go:130]:
```suggestion
		"stats.remote_cache.encrypt_enabled.count": 1,
```

File: pkg/infra/remotecache/remotecache.go [hunk 3]
```diff
@@ -72,12 +86,12 @@ type CacheStorage interface {
 
 	// Count returns the number of items in the cache.
 	// Optionaly a prefix can be provided to only count items with that prefix
+	// DO NOT USE. Not available for memcached.
```
Comment (sakjur) [inline pkg/infra/remotecache/remotecache.go:89]: Would it be a problem to remove count from the interface and wherever it’s used, use a temporary interface to check if it’s implemented?

```go
type countable interface {
	Count(ctx context.Context, prefix string) (int64, error)
}

func fun() {
…
if counter, countable := remoteCache; countable {
county, err := counter.Count(countext, "monte-cristo")
…
}
…
}
```

Comment (Jguer) [reply pkg/infra/remotecache/remotecache.go:89]: :+1: Seems good to me @sakjur , would make it safer even

**Pull request merged**